### DISCUSSION TITLE
Dexcom G6 and Xdrip

### DISCUSSION THREAD

**Discussion by Curly815** [NightscoutFoundation/xDrip#2038] [Q&A]
![Screenshot_20220306-113143_xDrip+](https://user-images.githubusercontent.com/71608093/156939096-cb3d2d72-41f8-4b11-872e-5380f3ae135a.jpg)
hey everyone! So Dex is my collector (because my doctors wants to be able to see clarity) and Xdrip is set as a share follower. It works well but then at times the signal will miss and be out for like 20 min or longer at times (like at this very moment signal has been missed for 21min) and i have noticed that signal misses when i lock my phone. I have checked all the settings and permissions on my phone to make sure Xdrip isnt killed when closing app and pretty sure all the settings on Xdrip are correct. I have notice a slight improvement if i leave the app open and not kill it. I also notice that if i got to the status tab and restart collector, 9 out of 10 times it refreshes with updated data. Not sire if this has been addressed and this is my first post so if this is repetitive, my bad! Any advice or suggestions would be great. Thanks and i love the Xdrip app! has been a total life saver! 

Here is a screen shot of the log. Thanks!!!

**Comment by Navid200** [comment]
Please have a look at this issue:
https://github.com/NightscoutFoundation/xDrip/issues/1849

Is it the same issue?

**Reply by Curly815** [reply]
yea it sounds like it but i dont understand what the actual solution was. 

I was looking into the whole hightscoutm heroku mango thing to see how that would work

**Reply by Navid200** [reply]
PR stands for Pull Request.
It means someone has proposed a solution.  But, it has not been approved yet.

**Reply by Curly815** [reply]
ohhh ok. Sorry, i am new to Github. That nightscout set up sounds complicated, lol. So i guess i kinda just hang and wait for solution. Is there a way to submit suggestions? i have a a couple that could be beneficial to Xdrip

**Reply by Navid200** [reply]
If you have a suggestion, you can open a thread just like when you made the first post here.
Choose an appropriate title.
In the post, explain the problem and how your suggestion will be beneficial.
Then, answer the  questions that may be posted.

Or, someone may add a link to another discussion or an issue and ask you if it is the same.  If it is the same, you can post in the existing thread if you think it will be better to keep everything in one place.

On the other hand, there are times you feel a thread has gone a long time and has taken a wrong turn.  You can always open a new thread.  No one can complain why you opened a thread.

**Reply by Curly815** [reply]
ok cool, thanks!!!

**Reply by Curly815** [reply]
has there even been a fix yet for having xdrip being able to upload to dexcom servers? i know it was working back in the day and then it all went to crap.

**Reply by Navid200** [reply]
xDrip can download from Dex share servers.

What do you mean by upload?

If you want to share from xDrip, you can use xDrip sync or Nightscout.

Please clarify exactly what you are talking about.

**Reply by Curly815** [reply]
So my endo wants to use clarity. When i setup xdrip as the collector and inut my id and pw for dexcom, it was not uploading to clarity (in the past, like a year ago maybe). So would i need nightscout for that? or should i be able to try it again

**Reply by Navid200** [reply]
As far as I know, xDrip has never been allowed to upload to Clarity.
You can either use a Dexcom receiver or a Dexcom app to do that.

**Reply by Curly815** [reply]
so then the best setup would be to have xdrip with the share follower setting right?

**Reply by Navid200** [reply]
If you Dexcom app or receiver is your collector, and you still want to use xDrip, yes, you need to set it up as a follower using Dex share servers.

**Reply by Curly815** [reply]
Exactly, which bring me to the issue of the random missed signal errors and having to restart collector multiple times to get it to go and if anyone on the team has identified a fix besides that one post you directed me too.

**Reply by Navid200** [reply]
I think we already went over this.
I stated already that there is a proposal, which has not been approved yet.

**Reply by Curly815** [reply]
i think we did also. sorry about that and thanks

**Comment by Navid200** [comment]
I thought the problem was that the readings would be delayed.  Not that it would stop until you restart collector.  
Does it stop?

**Reply by Curly815** [reply]
I think you are completely misunderstanding me. On xdrip, I get the error signal missed. All the dexcom stuff works fine. It's when xdrip tries to collect the data and it doesn't and it causes a signal miss. That's why I shared a screenshot of the event log.
